legions/commands/more/teatime.py

```diff
@@ -0,0 +1,309 @@
+        "eth1/GethTxPoolInspection",
+        "eth1/GethTxPoolStatus",
+        "eth1/GethStartRPC",
+        "eth1/GethStopRPC",
+    ]
+    + SUPPORTED_BY_ALL_CLIENTS,
+    NodeType.PARITY: [
+        "eth1/ParityGasCeiling",
+        "eth1/ParityGasFloor",
+        "eth1/ParityDevLogs",
+        "eth1/ParityChangeCoinbase",
+        "eth1/ParityChangeTarget",
+        "eth1/ParityChangeExtra",
+        "eth1/ParitySyncMode",
+        "eth1/ParityDropPeers",
+        "eth1/ParityUpgrade",
+        "eth1/ParityTxPoolStatistics",
+        "eth1/ParityTxCeiling",
+        "eth1/ParityMinGasPrice",
+    ]
+    + SUPPORTED_BY_ALL_CLIENTS,
+}
+
+
+@command
+class scan:
+    "RPC scans for blockchain nodes powered by teatime"
+
+    # The following variables are used as arguments for some of teatime's
+    # plugins.
+    infura_url = INFURA_URL
+
+    # Used when importing/creating a new account on the node. The accounts are
+    # locked with the specified password.
+    password = "legions"
+
+    # Used when trying to set a new gas ceiling target for mined blocks.
+    # Parity defaults to 0x0, see
+    # https://openethereum.github.io/wiki/JSONRPC-parity_set-module#parity_setgasceiltarget
+    gasceiling = "0x0"
+
+    # Used when trying to set a new gas floor target for mined blocks.
+    # Parity defaults to 0x0, see
+    # https://openethereum.github.io/wiki/JSONRPC-parity_set-module#parity_setgasfloortarget
+    gasfloor = "0x0"
+
+    # Used when trying to set the minimum transaction gas price.
+    gas_price = "default"
+
+    # Used when trying to set the maximum transaction gas.
+    gas_limit = "default"
+
+    # Used when trying to change the address of the author of the block
+    # (the beneficiary to whom the mining rewards were given).
+    author = "0x0"
+
+    # Used when trying to change the target chain.
+    target_chain = "foundation"  # Ethereum mainnet
+
+    # Used when trying to change the extra data field.
+    extra_data = "0x0"
+
+    # Used when trying to change the node's sync mode.
+    mode = "active"
+
+    # Used when checking whether the node is mining.
+    should_mine = True
+
+    # Used when checking whether the node has a certain hash rate.
+    hash_rate = 712000
+
+    # Used when checking whether the node has a certain peer count.
+    minimum_peercount = 3
+
+    # Used when trying to unlock accounts.
+    word_list = [""]
+    skip_below = 0  # skip accounts below this balance
+
+    # Used when checking SHA3 consistency.
+    test_input = ""
+    test_output = "c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
+
+    # Used when trying to add a peer to a node's peer list.
+    test_enode = ""
+
+    # Used when checking the node's sync state.
+    block_threshold = 10
+
+    # pluginInstantiator is a dict mapping plugin names to a function
+    # instantiating and returning them.
+    pluginInstantiator = {
+        "eth1/AccountCreation": lambda arg=password: AccountCreation(arg),
```

Comment:
When passing the args in the instantiator dict, we should use named parameters. That makes it easier to find bugs that result from plugin interface changes because Python will complain that the given parameter name is unexpected.